app/src/sharedTest/java/org/oppia/android/app/parser/CustomBulletSpanTest.kt

```diff
@@ -0,0 +1,229 @@
+//    assertThat(getNoofCustomBulletSpans(spannableString1)).isEqualTo(0) //4
+    assertThat(getNoofUnderlineSpans(spannableString1)).isEqualTo(1) // 0
+    assertThat(getNoofBulletSpans(spannableString2)).isEqualTo(0) // 0
+    assertThat(getNoofCustomBulletSpans(spannableString2)).isEqualTo(4) // 4
+    assertThat(getNoofUnderlineSpans(spannableString2)).isEqualTo(1) // 0
+  }
+
+  @Test
+  fun customBulletSpan_testReplaceBulletSpan_customBulletSpans_RemainsSame() {
+    val spannableString1 = testStringWithCustomBulletSpan
+    val spannableString2 = CustomBulletSpan.replaceBulletSpan(spannableString1, context)
+    assertThat(getNoofBulletSpans(spannableString1)).isEqualTo(0)
+    assertThat(getNoofCustomBulletSpans(spannableString1)).isEqualTo(1)
+    assertThat(getNoofBulletSpans(spannableString2)).isEqualTo(0)
+    assertThat(getNoofCustomBulletSpans(spannableString2)).isEqualTo(1)
+  }
+
+  @Test
+  fun customBulletSpan_testLeadMargin_isComputedToProperlyIndentText() {
+    val bulletRadius = context.resources.getDimensionPixelSize(
+      org.oppia.android.util.R.dimen.bullet_radius
+    )
+    val spacingBeforeBullet = context.resources.getDimensionPixelSize(
+      org.oppia.android.util.R.dimen.spacing_before_bullet
+    )
+    val spacingBeforeText = context.resources.getDimensionPixelSize(
+      org.oppia.android.util.R.dimen.spacing_before_text
+    )
+    val expectedMargin = spacingBeforeBullet + spacingBeforeText + 2 * bulletRadius
+    val spannableString = SpannableStringBuilder(testStringWithBulletSpan)
+    val customBulletSpannble = CustomBulletSpan.replaceBulletSpan(spannableString, context)
+    val leadingMargin = customBulletSpannble.getSpans(
+      0,
+      spannableString.length,
+      CustomBulletSpan::class.java
+    )[0].getLeadingMargin(true)
+    assertThat(leadingMargin).isEqualTo(expectedMargin)
+  }
+
+  private fun getBulletSpans(spannableString: SpannableStringBuilder): Array<out BulletSpan> {
+    return spannableString.getSpans<BulletSpan>(
+      0,
+      spannableString.length
+    )
+  }
+
+  private fun getCustomBulletSpans(
+    spannableString: SpannableStringBuilder
+  ): Array<out CustomBulletSpan> {
+    return spannableString.getSpans<CustomBulletSpan>(
+      0,
+      spannableString.length
+    )
+  }
+
+  private fun getUnderlineSpans(spannableString: SpannableStringBuilder): Array<out UnderlineSpan> {
+    return spannableString.getSpans<UnderlineSpan>(
+      0,
+      spannableString.length
+    )
+  }
+
+  private fun getNoofBulletSpans(spannableString: SpannableStringBuilder): Int {
+    return getBulletSpans(spannableString).size
+  }
+
+  private fun getNoofCustomBulletSpans(spannableString: SpannableStringBuilder): Int {
+    return getCustomBulletSpans(spannableString).size
+  }
+
+  private fun getNoofUnderlineSpans(spannableString: SpannableStringBuilder): Int {
+    return getUnderlineSpans(spannableString).size
+  }
```

Comment:
Added this to get count this can be shortened by including in the above functions